File: src/main/java/org/voegtle/weatherwidget/preferences/OrderCriteria.java
package org.voegtle.weatherwidget.preferences;

public enum OrderCriteria {
  location("location"), temperature("temperature"), rain("rain"), humidity("humidity");

  private String key;

  OrderCriteria(String key) {
    this.key = key;
  }

  public static OrderCriteria byKey(String key) {
    for (OrderCriteria criteria : values()) {
      if (criteria.key.equals(key)) {
        return criteria;
      }
    }
    return null;
  }
}

File: app/src/main/java/org/voegtle/weatherwidget/preferences/OrderCriteria.kt
package org.voegtle.weatherwidget.preferences

enum class OrderCriteria(private val key: String) {
  default("default"), location("location"), temperature("temperature"), rain("rain"), humidity("humidity");

  companion object {

    fun byKey(key: String): OrderCriteria {
      return entries.firstOrNull { it.key == key } ?: default
    }

    fun byIndex(which: Int): OrderCriteria =
        when (which) {
          0-> default
          1 -> location
          2 -> temperature
          3 -> rain
          4 -> humidity
          else -> default
        }

    fun index(find: OrderCriteria): Int {
      return entries.indexOf(find)
    }
  }

}
